augurpy/read_load.py

```diff
@@ -16,21 +22,53 @@ def load(input: Union[AnnData, DataFrame], meta: DataFrame, label_col: str, cell
         cell_type_col: column of the meta DataFrame or the Anndata or matrix containing the cell type labels for each
             cell in the cell-by-gene expression matrix
 
-
     Returns:
-        Pandas DataFrame containing gene expression values (cells in rows, genes in columns), cell type and condition
+        Anndata object containing gene expression values (cells in rows, genes in columns) and cell type, label as obs
     """
-    pass
+    if isinstance(input, AnnData):
+        input.obs = input.obs.rename(columns={cell_type_col: "cell_type", label_col: "label"})
+        out = input
+
+    elif isinstance(input, DataFrame):
+        try:
+            cell_type = input[cell_type_col]
+            label = input[label_col]
+        except KeyError:
+            print("No column names matching cell_type_col and label_col. Looking in meta data.")
+            try:
+                cell_type = meta[cell_type_col]
+                label = meta[label_col]
+            except (KeyError, TypeError):
+                raise Exception("Missing label and / or cell type column.") from None
```

Comment:
Just out of interest, why do you suggest `from None` aka removing some of the stack trace?

Reply:
Just because it looks nicer, not sure what the norm is.

Reply:
Usually you keep the whole stack trace because it might help in debugging.

Reply:
Yes that makes sense too! Ill change it accordingly.

Reply:
Ah this was why:
`augurpy/read_load.py:42:17: B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling.  See https://docs.python.org/3/tutorial/errors.html#exception-chaining for details.`

changing it to:
`except (KeyError, TypeError):
                print("Missing label and / or cell type column.")
                raise`